### PR TITLE
Remove UDP related code from JavaScript

### PR DESCRIPTION
**Pull request by pepone**
This PR removes UDP-related code from JavaScript. Since we do not have a UDP transport and there are no plans to add one, there is no reason to keep this code.

### CODE REVIEW

File: js/src/Ice/ImplicitContext.js [hunk 1]
```diff
@@ -2,25 +2,147 @@
...
-require("../Ice/HashUtil");
-require("../Ice/ArrayUtil");
-require("../Ice/StreamHelpers");
-require("../Ice/LocalException");
+const Ice = require("../Ice/ModuleRegistry").Ice;
+
 require("../Ice/Current");
-const Ice = _ModuleRegistry.module("Ice");
+require("../Ice/LocalException");
+
```
Comment (pepone) [inline js/src/Ice/ImplicitContext.js:9]: Unrelated to UDP, just merged ImplicitContextI and ImplicitContext files.

File: js/src/Ice/Logger.js [hunk 2]
```diff
@@ -2,23 +2,96 @@
...
-require("../Ice/ArrayUtil");
-require("../Ice/StreamHelpers");
-const Ice = _ModuleRegistry.module("Ice");
-
-const Slice = Ice.Slice;
-/* slice2js browser-bundle-skip-end */
-/* slice2js browser-bundle-skip */
-exports.Ice = Ice;
-/* slice2js browser-bundle-skip-end */
+const Ice = require("../Ice/ModuleRegistry").Ice;
```
Comment (pepone) [inline js/src/Ice/Logger.js:5]: also unrelated to UDP, merged Logger.js and LoggerI.js files.

**Comment by bernardnormier** [comment]
Can you still unmarshal / marshal / parse a proxy with a UDP endpoint, without using opaque?

I think it's desirable to keep this support.

**Comment by pepone** [comment]
> As of Ice 3.7, can js parse a proxy with a udp endpoint?

No, we ignore UDP endpoints when parsing.

> "hello:udp -h localhost -p 10000:tcp -h localhost -p 10000"

Parsing this result in a proxy with a single TCP endpoint, in both 3.7 and this PR

**Comment by bernardnormier** [comment]
> Parsing this result in a proxy with a single TCP endpoint, in both 3.7 and this PR

Ok, I wrote a little test for C++ and JS. And indeed, in both the ignore unknown endpoints when parsing an endpoint string and log a warning. For example:
```
*** [1/2] Running js/Ice/proxy tests ***
[ running client/server test - 05/31/24 13:59:27 ]
testing stringToProxy... -! 5/31/2024, 13:59:27.746 warning: Proxy contains unknown endpoints: `foo -h localhost' `udp -h localhost -p 1234'
ok
```

**Review by externl — APPROVED** [review]

**Pull request merged**